src/NHibernate/Engine/TypedValue.cs

```diff
@@ -63,53 +80,66 @@ public class ParameterListComparer : IEqualityComparer<TypedValue>
 		{
 			public bool Equals(TypedValue x, TypedValue y)
 			{
-				if (y == null) return false;
+				if (ReferenceEquals(x, y))
+					return true;
+				if (x == null || y == null)
+					return false;
```

Comment:
With its current usage, `x` is never supposed to be `null`. But as the comparer class is publicly accessible, better still check this.